src/internal.rs

```diff
@@ -511,7 +511,7 @@ pub struct Rav1dFrameContext_frame_thread {
     pub next_tile_row: [AtomicI32; 2],
 
     /// Indexed using `t.b.y * f.b4_stride + t.b.x`.
-    pub b: Vec<Av1Block>,
+    pub b: Vec<Mutex<Av1Block>>,
```

Comment:
This increases memory usage by 25% (32 vs. 40 bytes).  I think it's better to make the fields atomic, as they're generally accessed individually from what I can remember.  And we also get rid of the `Mutex` overhead.  I can open an issue for this.

Reply:
I think it makes more sense to access the block as a whole rather than add atomicity inside. We don't access blocks concurrently afaict. I'll make this vec DisjointMut for now to avoid the memory bloat.

Reply:
I was about to work on making `Av1Block` use `enum`s, which is a noisy but not super complex change.  Should I hold off on that to avoid conflicts with this then?

Reply:
Yes please, don't do that, that's not necessary.

Reply:
Okay I can wait, but what do you mean it's not necessary?  We need to remove the `union`s.

Reply:
I'm sorry, I read that as atomics. I should read more carefully when I'm tired. Just make sure you're coordinating with @randomPoison to ensure you're not both planning to do that. I think it is best to wait until this lands though, since you haven't done it yet.